### PR TITLE
Improve test coverage

### LINKED ISSUE #37: Improve Test Coverage
The current test coverage stands at 46%, which is below the desired threshold for ensuring code reliability and stability. In order to improve the overall quality of the codebase and minimize the risk of bugs and regressions, it would be nice to enhance the test coverage to reach a target of 60% or higher.

image

### PR DESCRIPTION
**Pull request by lincmba**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Resolves #37 

**Engineer Checklist**

- [ ] I have written **Unit tests** for any new feature(s) and edge cases for
      bug fixes
- [ ] I have added documentation for any new feature(s) and configuration
      option(s) on the `README.md`
- [ ] I have run `mvn spotless:check` to check my code follows the project's
      style guide
- [ ] I have run `mvn clean test jacoco:report` to confirm the coverage report
      was generated at `plugins/target/site/jacoco/index.html`
- [ ] I ran `mvn clean package` right before creating this pull request.

### CODE REVIEW

File: plugins/src/main/java/org/smartregister/fhir/gateway/plugins/PractitionerDetailsEndpointHelper.java [hunk 1]
```diff
@@ -88,7 +88,7 @@ public Bundle getSupervisorPractitionerDetailsByKeycloakId(String keycloakUuid)
         return bundle;
     }
 
-    private Bundle getAttributedPractitionerDetailsByPractitioner(Practitioner practitioner) {
```
Comment (ndegwamartin) [inline plugins/src/main/java/org/smartregister/fhir/gateway/plugins/PractitionerDetailsEndpointHelper.java:91]: We can change this to `protected` and it should still work, it has less visibility. 

Also add the annotation @VisibleForTesting

File: plugins/src/main/java/org/smartregister/fhir/gateway/plugins/PractitionerDetailsEndpointHelper.java [hunk 2]
```diff
@@ -189,7 +189,7 @@ public static Set<String> getAttributedLocations(List<LocationHierarchy> locatio
                 .collect(Collectors.toSet());
     }
 
-    private List<String> getOrganizationIdsByLocationIds(Set<String> attributedLocationsList) {
```
Comment (ndegwamartin) [inline plugins/src/main/java/org/smartregister/fhir/gateway/plugins/PractitionerDetailsEndpointHelper.java:192]: Same as above

File: plugins/src/main/java/org/smartregister/fhir/gateway/plugins/PractitionerDetailsEndpointHelper.java [hunk 3]
```diff
@@ -376,7 +376,7 @@ private Set<String> getOrganizationIdsByPractitionerRoles(
                 .collect(Collectors.toSet());
     }
 
-    private Practitioner getPractitionerByIdentifier(String identifier) {
```
Comment (ndegwamartin) [inline plugins/src/main/java/org/smartregister/fhir/gateway/plugins/PractitionerDetailsEndpointHelper.java:379]: SAB

File: plugins/src/main/java/org/smartregister/fhir/gateway/plugins/PractitionerDetailsEndpointHelper.java [hunk 4]
```diff
@@ -391,7 +391,7 @@ private Practitioner getPractitionerByIdentifier(String identifier) {
                 : null;
     }
 
-    private List<CareTeam> getCareTeamsByOrganizationIds(List<String> organizationIds) {
```
Comment (ndegwamartin) [inline plugins/src/main/java/org/smartregister/fhir/gateway/plugins/PractitionerDetailsEndpointHelper.java:394]: SAB

**Review by dubdabasoduba — APPROVED** [review]
